code/game/objects/structures/signs.dm

```diff
@@ -23,6 +28,48 @@
 		if(WEST)
 			pixel_x = -30
 
+//automatically adjust place and offset to make sure sign isn't floating in the middle of nowhere
+/obj/structure/sign/proc/place_poster()
+	var/turf/closed/applied_turf = get_step(loc, dir)
+	if(istype(applied_turf))
+		switch(dir)
+			if(NORTH)
+				pixel_y = 32
+			if(SOUTH)
+				pixel_y = -32
+			if(EAST)
+				pixel_x = 30
+			if(WEST)
+				pixel_x = -30
+		return
+	var/turf/locatedturf = get_turf(loc)
+	if(isclosedturf(locatedturf))
```

Comment:
we don't actually need to typecaste any of these turf vars. this one can be ```if(isclosedturf(get_turf(loc)))``` for example